bitbucket_server_test.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/jfrog/frogbot/utils"
+	"github.com/jfrog/froggit-go/vcsclient"
+	"github.com/jfrog/froggit-go/vcsutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io"
+	"net/http"
+	"testing"
+)
+
+const (
+	bitbucketServerIntegrationTokenEnv = "FROGBOT_TESTS_BB_SERVER_TOKEN"
+	bitbucketServerApiEndpoint         = "http://localhost:7990/rest"
+	bitbucketServerGitCloneUrl         = "http://localhost:7990/scm/frog/integration.git"
+)
+
+func buildBitbucketServerClient(t *testing.T, bitbucketServerToken string) vcsclient.VcsClient {
+	bbClient, err := vcsclient.NewClientBuilder(vcsutils.BitbucketServer).Username("frogbot").ApiEndpoint(bitbucketServerApiEndpoint).Token(bitbucketServerToken).Build()
+	assert.NoError(t, err)
+	return bbClient
+}
+
+func buildBitbucketServerIntegrationTestDetails(t *testing.T) *IntegrationTestDetails {
+	integrationRepoToken := getIntegrationToken(t, bitbucketServerIntegrationTokenEnv)
+	testDetails := NewIntegrationTestDetails(integrationRepoToken, string(utils.BitbucketServer), bitbucketServerGitCloneUrl, "FROG")
+	testDetails.ApiEndpoint = bitbucketServerApiEndpoint
+	return testDetails
+}
+
+func waitForConnection(t *testing.T) {
```

Comment:
I suggest adding the following info log message at the beginning of this function:
Waiting for Bitbucket Server to be up and running...